tools/superdsm/superdsm.xml

```diff
@@ -1,67 +1,76 @@
+    <description>with SuperDSM</description>
     <edam_operations>
         <edam_operation>operation_3443</edam_operation>
     </edam_operations>
-   <requirements> 
+    <xrefs>
+        <xref type="bio.tools">superdsm</xref>
+        <xref type="biii">superdsm</xref>
+    </xrefs>
+    <requirements> 
         <requirement type="package" version="0.1.3">superdsm</requirement>
         <requirement type="package" version="1.6.0">ray-core</requirement>
         <requirement type="package" version="0.18.1">scikit-image</requirement>
-   </requirements>
-   <command detect_errors="aggressive">
-   <![CDATA[
-   python '$__tool_directory__/run-superdsm.py'
-   '${dataset}'
-   'cfg.json'
-   'masks.png'
-   'overlay.png'
-   $seg_border
-   \${GALAXY_SLOTS:-4}
-   #if str($config.AF_scale) != '':
-       --AF_scale '${config.AF_scale}'
-   #end if
-   #if str($config.c2f_region_analysis_min_atom_radius) != '':
-       --c2f_region_analysis_min_atom_radius '${config.c2f_region_analysis_min_atom_radius}'
-   #end if
-   #if str($config.c2f_region_analysis_min_norm_energy_improvement) != '':
-       --c2f_region_analysis_min_norm_energy_improvement '${config.c2f_region_analysis_min_norm_energy_improvement}'
-   #end if
-   #if str($config.c2f_region_analysis_max_atom_norm_energy) != '':
-       --c2f_region_analysis_max_atom_norm_energy '${config.c2f_region_analysis_max_atom_norm_energy}'
-   #end if
-   #if str($config.c2f_region_analysis_max_cluster_marker_irregularity) != '':
-       --c2f_region_analysis_max_cluster_marker_irregularity '${config.c2f_region_analysis_max_cluster_marker_irregularity}'
-   #end if
-   #if str($config.dsm_alpha) != '':
-       --dsm_alpha '${config.dsm_alpha}'
-   #end if
-   #if str($config.dsm_AF_alpha) != '':
-       --dsm_AF_alpha '${config.dsm_AF_alpha}'
-   #end if
-   #if str($config.global_energy_minimization_beta) != '':
-       --global_energy_minimization_beta '${config.global_energy_minimization_beta}'
-   #end if
-   #if str($config.global_energy_minimization_AF_beta) != '':
-       --global_energy_minimization_AF_beta '${config.global_energy_minimization_AF_beta}'
-   #end if
-   #if str($config.postprocess_mask_max_distance) != '':
-       --postprocess_mask_max_distance '${config.postprocess_mask_max_distance}'
-   #end if
-   #if str($config.postprocess_mask_stdamp) != '':
-       --postprocess_mask_stdamp '${config.postprocess_mask_stdamp}'
-   #end if
-   #if str($config.postprocess_max_norm_energy) != '':
-       --postprocess_max_norm_energy '${config.postprocess_max_norm_energy}'
-   #end if
-   #if str($config.postprocess_min_contrast) != '':
-       --postprocess_min_contrast '${config.postprocess_min_contrast}'
-   #end if
-   #if str($config.postprocess_min_object_radius) != '':
-       --postprocess_min_object_radius '${config.postprocess_min_object_radius}'
-   #end if
-   ]]>
-   </command>
-   <inputs>
+        <requirement type="package" version="2020.0">mkl</requirement>

```

Comment:
do we really need to specify blas?

Reply:
I ran several tests on different hardware, including AMD and Intel CPUs, and found that using MKL always yields the fastest results, with even significant differences on some AMD Hardware: https://stackoverflow.com/questions/62783262/why-is-numpy-with-ryzen-threadripper-so-much-slower-than-xeon